shell/src/main/java/alluxio/cli/fs/command/CpCommand.java

```diff
@@ -241,8 +249,11 @@ public int run(CommandLine cl) throws AlluxioException, IOException {
         }
         CopyThreadPoolExecutor pool = new CopyThreadPoolExecutor(numThreads, System.out,
             mFileSystem.exists(dstPath) ? null : dstPath);
-        copyFromLocalFileList(pool, srcPaths, dstPath);
-        pool.shutdown();
+        try {
+          copyFromLocalFileList(pool, srcPaths, dstPath);
```

Comment:
in case copyFromLocalFileList fails for some reason, do you display exception / the list of files that failed somewhere?

Reply:
@madanadit After talking to @aaudiber, I changed the exception handling:

When exceptions are thrown during copying a file or traversing directory, copyFromLocal will print an error message to stderr, but will continue copying other files in the best effort. 

After all copy tasks succeed or fail, a summary of all errors including stacktraces will be printed out at the bottom of the command's output.